atomic_reactor/plugin.py

```diff
@@ -354,6 +381,57 @@ def __init__(self, dt, workflow, plugins_conf, *args, **kwargs):
         self.plugins_results = workflow.prebuild_results
         super(PreBuildPluginsRunner, self).__init__(dt, workflow, 'PreBuildPlugin', plugins_conf, *args, **kwargs)
 
+
+class BuildStepPlugin(BuildPlugin):
+    pass
+
+
+class BuildStepPluginsRunner(BuildPluginsRunner):
+
+    def __init__(self, dt, workflow, plugin_conf, *args, **kwargs):
+        self._build_result = None
+
+        logger.info("initializing runner of build-step plugin")
+        self.plugins_results = workflow.buildstep_result
+
+        if not plugin_conf:
+            plugin_conf = [{'name': 'docker_api'}]
+        else:
+            for i in range(len(plugin_conf)):
+                plugin_conf[i]['required'] = False
```

Comment:
How about a comment explaining why we're doing this (i.e. any build plugin which cannot be instantiated must be skipped without error).